workflow-manager/src/workflow_update.rs

```diff
@@ -0,0 +1,284 @@
+use std::collections::{BTreeMap, VecDeque};
+
+use anyhow::Context;
+use cosmwasm_std::{to_json_binary, CosmosMsg, WasmMsg};
+use cw_ownable::Expiration;
+use serde::{Deserialize, Serialize};
+use valence_authorization_utils::{
+    authorization::{AuthorizationInfo, AuthorizationModeInfo, Priority},
+    authorization_message::{Message, MessageDetails, MessageType},
+    builders::{AtomicActionBuilder, AtomicActionsConfigBuilder, AuthorizationBuilder},
+    msg::ProcessorMessage,
+};
+use valence_service_utils::{Id, ServiceAccountType};
+
+use crate::{
+    connectors::Connectors,
+    domain::Domain,
+    error::{ManagerError, ManagerResult},
+    service::ServiceConfigUpdate,
+    NEUTRON_CHAIN,
+};
+
+/// The job of the update, is to output a set of instructions to the user to update the workflow configuration.  
+/// The user can only update service configs and authorizations.
+
+#[derive(Clone, Debug, Default, Serialize, Deserialize)]
+pub struct WorkflowConfigUpdate {
+    /// This is the id of the workflow
+    /// Required for update
+    pub id: u64,
+    /// New owner, if the owner is to be updated
+    pub owner: Option<String>,
+    /// The list service data by id
+    pub services: BTreeMap<Id, ServiceConfigUpdate>,
+    /// A list of authorizations
+    pub authorizations: Vec<AuthorizationInfoUpdate>,
+}
+
+#[derive(Clone, Debug, Serialize, Deserialize)]
+pub enum AuthorizationInfoUpdate {
+    Add(AuthorizationInfo),
+    Modify {
+        label: String,
+        not_before: Option<Expiration>,
+        expiration: Option<Expiration>,
+        max_concurrent_executions: Option<u64>,
+        priority: Option<Priority>,
+    },
+    /// Disable by label
+    Disable(String),
+    /// Disable by label
+    Enable(String),
+}
+
+#[derive(Clone, Debug)]
+pub struct UpdateResponse {
+    pub instructions: Vec<CosmosMsg>,
+}
+
+impl WorkflowConfigUpdate {
+    /// Modify an existing config with a new config
+    pub async fn update(&mut self, connectors: &Connectors) -> ManagerResult<UpdateResponse> {
+        let neutron_domain = Domain::CosmosCosmwasm(NEUTRON_CHAIN.to_string());
+
+        // get the old workflow config from registry
+        let mut neutron_connector = connectors.get_or_create_connector(&neutron_domain).await?;
+
+        if self.id == 0 {
+            return Err(ManagerError::IdIsZero);
```

Comment:
A bit of a weird error .. Maybe you could create a constructor where you enforce that it's not possible to create WorkflowConfigUpdate with an `id` equal to zero (and any other basic validation).

Reply:
Maybe something more explicit like `InvalidWorkflowId` ?